In [2]:
import pandas as pd

data = pd.read_json('02_gls_raw_all.json')

data.info()
data.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4407 entries, 0 to 4406
Data columns (total 26 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   address           4407 non-null   object 
 1   pclshopid         4407 non-null   object 
 2   _leaflet_id       4407 non-null   object 
 3   ctrcode           4407 non-null   object 
 4   city              4407 non-null   object 
 5   geolat            4407 non-null   float64
 6   geolng            4407 non-null   float64
 7   iscodhandler      4407 non-null   int64  
 8   isparcellocker    4407 non-null   int64  
 9   contact           4407 non-null   object 
 10  phone             4407 non-null   object 
 11  name              4407 non-null   object 
 12  guid              4407 non-null   object 
 13  visible           4407 non-null   int64  
 14  utilization       4407 non-null   int64  
 15  capacitystate     4407 non-null   int64  
 16  parcelBox         4407 non-null   int64  


geolat       geolng  iscodhandler  isparcellocker  visible  \
count  4407.000000  4407.000000        4407.0     4407.000000   4407.0   
mean     49.865015    15.365971           1.0        0.617200      1.0   
std       0.497417     1.556631           0.0        0.486125      0.0   
min      48.616100    12.167782           1.0        0.000000      1.0   
25%      49.506227    14.338150           1.0        0.000000      1.0   
50%      49.954936    14.847244           1.0        1.000000      1.0   
75%      50.156023    16.615144           1.0        1.000000      1.0   
max      51.018583    18.823819           1.0        1.000000      1.0   

       utilization  capacitystate    parcelBox   parcelShop      payCard  \
count  4407.000000    4407.000000  4407.000000  4407.000000  4407.000000   
mean      8.463581       1.545496     0.617200     0.382800     0.369412   
std      23.847735       1.527820     0.486125     0.486125     0.482700   
min       0.000000       1.000000     0.000000     0.000000     0.000000   
25%       0.000000       1.000000     0.000000     0.000000     0.000000   
50%       0.000000       1.000000     1.000000     0.000000     0.000000   
75%       0.000000       1.000000     1.000000     1.000000     1.000000   
max      94.000000      10.000000     1.000000     1.000000     1.000000   

       paybybankcard       zipcode  openSaturday   openSunday  
count    4407.000000   4407.000000   4407.000000  4407.000000  
mean        0.369412  45205.354436      0.810983     0.660994  
std         0.482700  20771.713109      0.391567     0.473426  
min         0.000000  10000.000000      0.000000     0.000000  
25%         0.000000  27601.000000      1.000000     0.000000  
50%         0.000000  41501.000000      1.000000     1.000000  
75%         1.000000  66454.500000      1.000000     1.000000  
max         1.000000  79852.000000      1.000000     1.000000

In [3]:
df = pd.DataFrame(data)

df.head(2)

address             pclshopid           _leaflet_id ctrcode  \
0  S. K. Neumanna 590/22  CZ58601-PARCELLOCK01  CZ58601-PARCELLOCK01      CZ   
1       Myslotínská 2449         39301-ELPESRO         39301-ELPESRO      CZ   

        city   geolat  geolng  iscodhandler  isparcellocker contact  ...  \
0    Jihlava  49.4048  15.557             1               1          ...   
1  Pelhřimov  49.4139  15.220             1               0          ...   

  parcelBox parcelShop payCard  paybybankcard  wheelchairAccess  zipcode  \
0         1          0       1              1                      58601   
1         0          1       1              1                      39301   

   openSaturday  openSunday  \
0             1           1   
1             1           0   

                                            imageUrl  holiday  
0  https://prodpdmpublicstore.blob.core.windows.n...      NaN  
1  https://prodpdmpublicstore.blob.core.windows.n...      NaN  

[2 rows x 26 columns]

In [4]:
df.shape

(4407, 26)

Dotazy:
Jak si na začátku vyobrazím sloupce, které neobsahují nic, není to NaN

Vytvořím si pomocný dataset csv soubor. Lépe se mi kontroluje postup v csv. 
Se souborem pracuji průběžně, na konci procesu jej mažu.

Vybrat v DF sloupce pro lepší přehlednost. Tam, kde je 100% jistota.
Zbývající nechám pro další investigaci. Kontroluji DF, ale následně ještě
prověřuji i v csv.

Vyobrazit sloupce, které neobsahují nic.
Pokus - vybrat sloupce, které nejsou NaN. Nefunguje na "bílé místo".

In [5]:
df.to_csv('gls_pomocny_dataset.csv', index=False, encoding='utf-8')

Pro jistotu vynutím u zipcode string

In [6]:
df.loc[:,'zipcode'] = df['zipcode'].astype(str)

C:\Users\Sabina\AppData\Local\Temp\ipykernel_25296\3660544216.py:1: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '['58601' '39301' '16000' ... '46007' '11000' '47125']' has dtype incompatible with int64, please explicitly cast to a compatible dtype first.
  df.loc[:,'zipcode'] = df['zipcode'].astype(str)


In [7]:
not_null_columns = df.columns[df.notna().all()]
not_null_columns

Index(['address', 'pclshopid', '_leaflet_id', 'ctrcode', 'city', 'geolat',
       'geolng', 'iscodhandler', 'isparcellocker', 'contact', 'phone', 'name',
       'guid', 'visible', 'utilization', 'capacitystate', 'parcelBox',
       'parcelShop', 'payCard', 'paybybankcard', 'wheelchairAccess', 'zipcode',
       'openSaturday', 'openSunday', 'imageUrl'],
      dtype='object')

In [8]:
null_columns = df.columns[df.isnull().all()]
null_columns


Index([], dtype='object')

V csv vyberu sloupce, které neobsahují nic, nebo je nepotřebuji, nebo nic nefiltrují
např. visible, iscodhandler obsahuje kód 1 pro celý dataset.
axis=1 - horizontálně - odstraní sloupce

In [9]:
k_odstraneni = ['contact', 'phone', 'guid', 'visible', 'payCard', 'paybybankcard', 'wheelchairAccess', 'imageUrl', 'holiday', 'iscodhandler']
df = df.drop(k_odstraneni, axis=1)

df.columns
df.head(2) 

address             pclshopid           _leaflet_id ctrcode  \
0  S. K. Neumanna 590/22  CZ58601-PARCELLOCK01  CZ58601-PARCELLOCK01      CZ   
1       Myslotínská 2449         39301-ELPESRO         39301-ELPESRO      CZ   

        city   geolat  geolng  isparcellocker  \
0    Jihlava  49.4048  15.557               1   
1  Pelhřimov  49.4139  15.220               0   

                               name  utilization  capacitystate  parcelBox  \
0                    GLS Parcel Box            0              1          1   
1  Elpe s.r.o. - Maloobchod elektro            0              1          0   

   parcelShop zipcode  openSaturday  openSunday  
0           0   58601             1           1  
1           1   39301             1           0

Sloupec ISPARCELLOCKER  1 = BOX,  0 = výdejní místo nebo provozovna, depo
Sloupec Parcelbox       1 = BOX,  0 = výdejní místo nebo provozovna, depo
Sloupec ParcelShop      1 = BOX,  0 = výdejní místo nebo provozovna, depo
kontrola v mapě na webu gls.
Vymazat provozovny/pobočky, vybrat jen boxy

In [10]:
boxy = df[df['isparcellocker'] == 1]

boxy

address             pclshopid           _leaflet_id  \
0     S. K. Neumanna 590/22  CZ58601-PARCELLOCK01  CZ58601-PARCELLOCK01   
743       Sokolovská 147/25    CZ59401-PARCELLOCK    CZ59401-PARCELLOCK   
747            Králova 1256    CZ75701-PARCELLOCK    CZ75701-PARCELLOCK   
748      Pod Třešňovkou 234  CZ15600-PARCELLOCK01  CZ15600-PARCELLOCK01   
749           Svisle 1100/9    CZ75002-PARCELLOCK    CZ75002-PARCELLOCK   
...                     ...                   ...                   ...   
4373      Litoměřická brána    CZ41201-PLOCKER007    CZ41201-PLOCKER007   
4374               Na Hrázi    CZ66701-PLOCKER003    CZ66701-PLOCKER003   
4376            Cejl 42/109    CZ60200-PLOCKER020    CZ60200-PLOCKER020   
4396    Valchařská 1657/41b    CZ61400-PLOCKER007    CZ61400-PLOCKER007   
4406                Lužická    CZ47125-PLOCKER002    CZ47125-PLOCKER002   

     ctrcode                       city     geolat     geolng  isparcellocker  \
0         CZ                    Jihlava  49.404800  15.557000               1   
743       CZ             Velké Meziříčí  49.351700  16.019600               1   
747       CZ          Valašské Meziříčí  49.469300  17.971100               1   
748       CZ                      Praha  49.970800  14.383200               1   
749       CZ                     Přerov  49.450300  17.459400               1   
...      ...                        ...        ...        ...             ...   
4373      CZ                    Terezín  50.513728  14.135369               1   
4374      CZ               Židlochovice  49.043358  16.616224               1   
4376      CZ          Brno - Zábrdovice  49.201348  16.625946               1   
4396      CZ  Brno-Husovice x) (Brno-me  49.215403  16.639576               1   
4406      CZ      Jablonné v Podještědí  50.771413  14.768869               1   

                                                 name  utilization  \
0                                      GLS Parcel Box            0   
743                                    GLS Parcel Box            0   
747                                    GLS Parcel Box            0   
748                                    GLS Parcel Box            0   
749                                    GLS Parcel Box            0   
...                                               ...          ...   
4373            AlzaBox Terezín - Retail Park (Pepco)            0   
4374                      AlzaBox Židlochovice (Lidl)            0   
4376                                   GLS Parcel Box            0   
4396             AlzaBox Brno - Husovice - Valchařská            0   
4406  AlzaBox Jablonné v Podještědí - Lužická (Penny)            0   

      capacitystate  parcelBox  parcelShop zipcode  openSaturday  openSunday  
0                 1          1           0   58601             1           1  
743               1          1           0   59401             1           1  
747               1          1           0   75701             1           1  
748               1          1           0   15600             1           1  
749               1          1           0   75002             1           1  
...             ...        ...         ...     ...           ...         ...  
4373              1          1           0   41201             1           1  
4374              1          1           0   66701             1           1  
4376              1          1           0   60200             1           1  
4396              1          1           0   61400             1           1  
4406              1          1           0   47125             1           1  

[2720 rows x 16 columns]

Vybírám do dataframu jen boxy bez poboček

In [11]:
df = boxy


Kontrola, kolik záznamů zůstalo v seznamu

In [12]:
df.shape

(2720, 16)

V seznamu jsou boxy od 3. stran AlzaBox a AlzaWall (to je v Alzaprodejně)
| - regex nebo

In [13]:
alza_boxwall = df[df['name'].str.contains(r'AlzaBox|AlzaWall', na=False)]

df.columns
alza_boxwall

address             pclshopid           _leaflet_id ctrcode  \
877       Na Kamenci 2148    CZ51101-PARCELLOCK    CZ51101-PARCELLOCK      CZ   
878           Lorecká 655    CZ28401-PARCELLOCK    CZ28401-PARCELLOCK      CZ   
881            Sběrná 654  CZ28401-PARCELLOCK01  CZ28401-PARCELLOCK01      CZ   
882       U Mototechny 89    CZ25162-PARCELLOCK    CZ25162-PARCELLOCK      CZ   
883         Sousedská 600  CZ46001-PARCELLOCK02  CZ46001-PARCELLOCK02      CZ   
...                   ...                   ...                   ...     ...   
4372     Novoborská 610/2    CZ19000-PLOCKER017    CZ19000-PLOCKER017      CZ   
4373    Litoměřická brána    CZ41201-PLOCKER007    CZ41201-PLOCKER007      CZ   
4374             Na Hrázi    CZ66701-PLOCKER003    CZ66701-PLOCKER003      CZ   
4396  Valchařská 1657/41b    CZ61400-PLOCKER007    CZ61400-PLOCKER007      CZ   
4406              Lužická    CZ47125-PLOCKER002    CZ47125-PLOCKER002      CZ   

                           city     geolat     geolng  isparcellocker  \
877                      Turnov  50.580500  15.155400               1   
878                  Kutná Hora  49.954300  15.271000               1   
881                  Kutná Hora  49.957900  15.275900               1   
882                     Tehovec  49.991300  14.725100               1   
883                     Liberec  50.776700  15.029400               1   
...                         ...        ...        ...             ...   
4372                    PRAHA 9  50.124687  14.502355               1   
4373                    Terezín  50.513728  14.135369               1   
4374               Židlochovice  49.043358  16.616224               1   
4396  Brno-Husovice x) (Brno-me  49.215403  16.639576               1   
4406      Jablonné v Podještědí  50.771413  14.768869               1   

                                                  name  utilization  \
877                              AlzaBox Turnov (Lidl)           62   
878                         AlzaBox Kutná Hora (Billa)           86   
881                AlzaBox Kutná Hora - Sběrná (Penny)            0   
882        AlzaBox Mukařov - U Mototechny (Stavebniny)            0   
883                           AlzaBox Liberec (Globus)            0   
...                                                ...          ...   
4372  AlzaBox P9 - Prosek - Novoborská (Střední škola)            0   
4373             AlzaBox Terezín - Retail Park (Pepco)            0   
4374                       AlzaBox Židlochovice (Lidl)            0   
4396              AlzaBox Brno - Husovice - Valchařská            0   
4406   AlzaBox Jablonné v Podještědí - Lužická (Penny)            0   

      capacitystate  parcelBox  parcelShop zipcode  openSaturday  openSunday  
877               5          1           0   51101             1           1  
878               5          1           0   28401             1           1  
881               1          1           0   28401             1           1  
882               1          1           0   25162             1           1  
883               1          1           0   46001             1           1  
...             ...        ...         ...     ...           ...         ...  
4372              1          1           0   19000             1           1  
4373              1          1           0   41201             1           1  
4374              1          1           0   66701             1           1  
4396              1          1           0   61400             1           1  
4406              1          1           0   47125             1           1  

[1747 rows x 16 columns]

In [14]:
bez_alza_boxwall = df[~df['name'].str.contains(r'AlzaBox|AlzaWall', na=False)]

df.columns
bez_alza_boxwall

address             pclshopid           _leaflet_id  \
0      S. K. Neumanna 590/22  CZ58601-PARCELLOCK01  CZ58601-PARCELLOCK01   
743        Sokolovská 147/25    CZ59401-PARCELLOCK    CZ59401-PARCELLOCK   
747             Králova 1256    CZ75701-PARCELLOCK    CZ75701-PARCELLOCK   
748       Pod Třešňovkou 234  CZ15600-PARCELLOCK01  CZ15600-PARCELLOCK01   
749            Svisle 1100/9    CZ75002-PARCELLOCK    CZ75002-PARCELLOCK   
...                      ...                   ...                   ...   
4271     Ladislava Váchy 517    CZ76001-PLOCKER009    CZ76001-PLOCKER009   
4272        Langrova 1227/1d    CZ62700-PLOCKER003    CZ62700-PLOCKER003   
4301  Komenského náměstí 999    CZ59231-PLOCKER006    CZ59231-PLOCKER006   
4309         Zastávecká 1267    CZ66501-PLOCKER003    CZ66501-PLOCKER003   
4376             Cejl 42/109    CZ60200-PLOCKER020    CZ60200-PLOCKER020   

     ctrcode                  city     geolat     geolng  isparcellocker  \
0         CZ               Jihlava  49.404800  15.557000               1   
743       CZ        Velké Meziříčí  49.351700  16.019600               1   
747       CZ     Valašské Meziříčí  49.469300  17.971100               1   
748       CZ                 Praha  49.970800  14.383200               1   
749       CZ                Přerov  49.450300  17.459400               1   
...      ...                   ...        ...        ...             ...   
4271      CZ                  Zlín  49.217261  17.633535               1   
4272      CZ                  Brno  49.179369  16.679533               1   
4301      CZ  Nové Město na Moravě  49.563083  16.075644               1   
4309      CZ         Rosice u Brna  49.184816  16.368091               1   
4376      CZ     Brno - Zábrdovice  49.201348  16.625946               1   

                name  utilization  capacitystate  parcelBox  parcelShop  \
0     GLS Parcel Box            0              1          1           0   
743   GLS Parcel Box            0              1          1           0   
747   GLS Parcel Box            0              1          1           0   
748   GLS Parcel Box            0              1          1           0   
749   GLS Parcel Box            0              1          1           0   
...              ...          ...            ...        ...         ...   
4271  GLS Parcel Box            0              1          1           0   
4272  GLS Parcel Box            0              1          1           0   
4301  GLS Parcel Box            0              1          1           0   
4309  GLS Parcel Box            0              1          1           0   
4376  GLS Parcel Box            0              1          1           0   

     zipcode  openSaturday  openSunday  
0      58601             1           1  
743    59401             1           1  
747    75701             1           1  
748    15600             1           1  
749    75002             1           1  
...      ...           ...         ...  
4271   76001             1           1  
4272   62700             1           1  
4301   59231             1           1  
4309   66501             1           1  
4376   60200             1           1  

[973 rows x 16 columns]

Vybrat do DF jen řádky bez Alzabox a Alzawall. Nyní zůstane v datasetu pouze
vše, co je box a není to Alzabox a Alza wall. Zbývá 973 řádků

In [15]:
df = bez_alza_boxwall

In [16]:
df.shape

(973, 16)

pomocný krok - přepsat pomocné csv, opět lepší vizualizace

In [17]:
df.to_csv('gls_pomocny_dataset.csv', index=False, encoding='utf-8')

Vybrat sloupce do df
Zahodit otevirací dobu

In [18]:
df = df[['name', 'city', 'zipcode', 'address', 'geolat', 'geolng']]

df.columns
df.head(2)

name            city zipcode                address   geolat  \
0    GLS Parcel Box         Jihlava   58601  S. K. Neumanna 590/22  49.4048   
743  GLS Parcel Box  Velké Meziříčí   59401      Sokolovská 147/25  49.3517   

      geolng  
0    15.5570  
743  16.0196

Ukázka a odstranění "( TEXT)" na konci řádků ve sloupci address a name

In [19]:
adresy_se_zavorkou = df[df['address'].str.contains(r'\(.*\)$', na=False)]

adresy_se_zavorkou

name              city zipcode  \
752   GLS Parcel Box (NC Haná)           Olomouc   77900   
761             GLS Parcel Box           Havířov   73601   
766             GLS Parcel Box           Ostrava   72527   
776             GLS Parcel Box  České Budějovice   37001   
777             GLS Parcel Box              Brno   62500   
838             GLS Parcel Box             Praha   16900   
2358            GLS Parcel Box             Praha   17000   
2912            GLS Parcel Box     Brno Židenice   61500   
3069            GLS Parcel Box             Praha   14000   
3128            GLS Parcel Box            Hořice   50801   
3771            GLS Parcel Box       Králův Dvůr   26701   

                                          address     geolat     geolng  
752                      Kafkova 1223/8 (OC Haná)  49.575400  17.228200  
761                            Dělnická (OC Elan)  49.780800  18.431800  
766                      Spojovací 396 (NC Géčko)  49.849000  18.145800  
776                    Nádražní 1759 (OC Mercury)  48.972900  14.487400  
777      Netroufalky 770/14-16 (OC Campus Square)  49.177700  16.566800  
838   Chaloupeckého 1914/9 (hřiště proti bloku 4)  50.080300  14.392800  
2358               Veletržní 1623/24 (-1P Garáže)  50.101736  14.430196  
2912                    Bubeníčkova (stará osada)  49.201776  16.641489  
3069      Duhová 1531/3 (zadní strana parkoviště)  50.049215  14.450828  
3128                    Škroupova (vedle jednoty)  50.367666  15.635367  
3771                 náměstí Míru 139 (u radnice)  49.944745  14.039454

df.loc[pro všechny řádky, ve sloupci address]

In [20]:
df.loc[:, 'address'] = df['address'].str.replace(r'\s*\(.*\)$', '', regex=True)
adresy_se_zavorkou = df[df['address'].str.contains(r'\(.*\)$', na=False)]

df.columns
adresy_se_zavorkou

Empty DataFrame
Columns: [name, city, zipcode, address, geolat, geolng]
Index: []

In [21]:
df.loc[:, 'name'] = df['name'].str.replace(r'\s*\(.*\)$', '', regex=True)
name_se_zavorkou = df[df['name'].str.contains(r'\(.*\)$', na=False)]

name_se_zavorkou

Empty DataFrame
Columns: [name, city, zipcode, address, geolat, geolng]
Index: []

Ukázka a oddělení č.p. z pole address

In [22]:
adresy_s_cislem = df[df['address'].str.contains(r'\s\d[\dA-Za-z/,\- ]*(?:/[IVXLCDM]+)?$', na=False)]

df.columns
adresy_s_cislem

name                  city zipcode                 address  \
0     GLS Parcel Box               Jihlava   58601   S. K. Neumanna 590/22   
743   GLS Parcel Box        Velké Meziříčí   59401       Sokolovská 147/25   
747   GLS Parcel Box     Valašské Meziříčí   75701            Králova 1256   
748   GLS Parcel Box                 Praha   15600      Pod Třešňovkou 234   
749   GLS Parcel Box                Přerov   75002           Svisle 1100/9   
...              ...                   ...     ...                     ...   
4271  GLS Parcel Box                  Zlín   76001     Ladislava Váchy 517   
4272  GLS Parcel Box                  Brno   62700        Langrova 1227/1d   
4301  GLS Parcel Box  Nové Město na Moravě   59231  Komenského náměstí 999   
4309  GLS Parcel Box         Rosice u Brna   66501         Zastávecká 1267   
4376  GLS Parcel Box     Brno - Zábrdovice   60200             Cejl 42/109   

         geolat     geolng  
0     49.404800  15.557000  
743   49.351700  16.019600  
747   49.469300  17.971100  
748   49.970800  14.383200  
749   49.450300  17.459400  
...         ...        ...  
4271  49.217261  17.633535  
4272  49.179369  16.679533  
4301  49.563083  16.075644  
4309  49.184816  16.368091  
4376  49.201348  16.625946  

[948 rows x 6 columns]

In [23]:
adresy_bez_cisla = df[~df['address'].str.contains(r'\s\d[\dA-Za-z/,\- ]*(?:/[IVXLCDM]+)?$', na=False)]

df.columns
adresy_bez_cisla

name                      city zipcode  \
761   GLS Parcel Box                   Havířov   73601   
763   GLS Parcel Box                      Brno   60200   
796   GLS Parcel Box                     Praha   14000   
1531  GLS Parcel Box                   Ostrava   70030   
1909  GLS Parcel Box                   Jeseník   79001   
2375  GLS Parcel Box              Uherský Brod   68801   
2649  GLS Parcel Box         LUKA NAD JIHLAVOU   58822   
2650  GLS Parcel Box                    Dačice   38001   
2659  GLS Parcel Box       Kralupy nad Vltavou   27801   
2864  GLS Parcel Box        Veselí nad Lužnicí   39181   
2868  GLS Parcel Box                   Liberec   46001   
2912  GLS Parcel Box             Brno Židenice   61500   
2917  GLS Parcel Box                    Bílina   41801   
2960  GLS Parcel Box                  Litomyšl   57001   
3096  GLS Parcel Box                Česká Lípa   47001   
3106  GLS Parcel Box                    Náchod   54701   
3128  GLS Parcel Box                    Hořice   50801   
3133  GLS Parcel Box       Rychnov nad Kněžnou   51601   
3207  GLS Parcel Box                     Praha   14700   
3395  GLS Parcel Box                  Frýdlant   46401   
3466  GLS Parcel Box                 Nová Paka   50901   
3532  GLS Parcel Box  Praha 20 Horní Počernice   19300   
3747  GLS Parcel Box                   Brtnice   58832   
3995  GLS Parcel Box        Chrást nad Sázavou   25742   
4050  GLS Parcel Box                    Kladno   27201   

                                                address     geolat     geolng  
761                                            Dělnická  49.780800  18.431800  
763                                    Mendlovo náměstí  49.190446  16.596150  
796                                     Na Hřebenech II  50.053585  14.430863  
1531                   Dr. Lukášové křižovatka Holasova  49.787700  18.256500  
1909                        Autobusové n.-Smetanova ul.  50.227500  17.208300  
2375                                            Močidla  49.025775  17.660271  
2649                                           Dělnická  49.377473  15.698793  
2650                                         Pantočkova  49.080330  15.430082  
2659                                     Libušina ulice  50.240607  14.317116  
2864            Tř. Československé Armády 609 - u myček  49.197283  14.716687  
2868                                      Náměstí Tržní  50.774111  15.060722  
2912                                        Bubeníčkova  49.201776  16.641489  
2917                      Aléská 269, Pražské předměstí  50.542161  13.776977  
2960                     Sokolovská 113 - za automyčkou  49.880838  16.292915  
3096                        Na Blatech - automyčka DIBO  50.687907  14.519294  
3106                               Mlýnská - parkoviště  50.414193  16.162277  
3128                                          Škroupova  50.367666  15.635367  
3133           Zborovská 17 - vedle katastrálního úřadu  50.163417  16.278833  
3207       Ke Krči - zastávka autobusu Branické náměstí  50.034667  14.414569  
3395                           Automyčka JOMI Fügnerova  50.923091  15.086144  
3466                         Na Tržišti 23 - Parkoviště  50.494030  15.515213  
3532       Pavlišovská 2292/14 - vjezd z ulice Jizbická  50.110422  14.621840  
3747                       parkoviště u ulice Sportovní  49.309596  15.668806  
3995                Chrást nad Sázavou 239 - parkoviště  49.847983  14.551752  
4050  Pařížská - vedle novostavby řadových garáží př...  50.135930  14.135830

extrahovat č.p. z address a poté vymazat z address

In [24]:
df.loc[:,'house_no'] = df['address'].str.extract(r'(\d[\dA-Za-z/,\- ]*(/[IVXLCDM]+)?)$')[0]

df.head(2)


C:\Users\Sabina\AppData\Local\Temp\ipykernel_25296\1560429831.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[:,'house_no'] = df['address'].str.extract(r'(\d[\dA-Za-z/,\- ]*(/[IVXLCDM]+)?)$')[0]


name            city zipcode                address   geolat  \
0    GLS Parcel Box         Jihlava   58601  S. K. Neumanna 590/22  49.4048   
743  GLS Parcel Box  Velké Meziříčí   59401      Sokolovská 147/25  49.3517   

      geolng house_no  
0    15.5570   590/22  
743  16.0196   147/25

In [25]:
df.loc[:,'address'] = df['address'].str.replace(r'\s\d[\dA-Za-z/,\- ]*(/[IVXLCDM]+)?$', '', regex = True)

df.head(2)

name            city zipcode         address   geolat   geolng  \
0    GLS Parcel Box         Jihlava   58601  S. K. Neumanna  49.4048  15.5570   
743  GLS Parcel Box  Velké Meziříčí   59401      Sokolovská  49.3517  16.0196   

    house_no  
0     590/22  
743   147/25

POMOCNÉ CSV

In [26]:
df.to_csv('gls_pomocny_dataset.csv', index=False, encoding='utf-8')

DOČIŠTĚNÍ CHYB v address
- před č. p. text za pomlčkou

In [27]:
adresy_s_pomlckou = df[df['address'].str.contains(r'-', na=False)]

df.columns
adresy_s_pomlckou

name                      city zipcode  \
1909  GLS Parcel Box                   Jeseník   79001   
2864  GLS Parcel Box        Veselí nad Lužnicí   39181   
2870  GLS Parcel Box                   Holešov   76901   
2960  GLS Parcel Box                  Litomyšl   57001   
3096  GLS Parcel Box                Česká Lípa   47001   
3106  GLS Parcel Box                    Náchod   54701   
3133  GLS Parcel Box       Rychnov nad Kněžnou   51601   
3207  GLS Parcel Box                     Praha   14700   
3466  GLS Parcel Box                 Nová Paka   50901   
3532  GLS Parcel Box  Praha 20 Horní Počernice   19300   
3775  GLS Parcel Box                     Praha   14800   
3995  GLS Parcel Box        Chrást nad Sázavou   25742   
4009  GLS Parcel Box                    Trmice   40004   
4050  GLS Parcel Box                    Kladno   27201   
4173  GLS Parcel Box       Moravské Budějovice   67541   
4262  GLS Parcel Box                     Praha   16300   
4264  GLS Parcel Box                     Praha   16100   

                                                address     geolat     geolng  \
1909                        Autobusové n.-Smetanova ul.  50.227500  17.208300   
2864            Tř. Československé Armády 609 - u myček  49.197283  14.716687   
2870                   Obchodní galerie - ulice Tovární  49.327873  17.573918   
2960                     Sokolovská 113 - za automyčkou  49.880838  16.292915   
3096                        Na Blatech - automyčka DIBO  50.687907  14.519294   
3106                               Mlýnská - parkoviště  50.414193  16.162277   
3133           Zborovská 17 - vedle katastrálního úřadu  50.163417  16.278833   
3207       Ke Krči - zastávka autobusu Branické náměstí  50.034667  14.414569   
3466                         Na Tržišti 23 - Parkoviště  50.494030  15.515213   
3532       Pavlišovská 2292/14 - vjezd z ulice Jizbická  50.110422  14.621840   
3775         zastávka Vídeňská - naproti K Zelené louce  50.020734  14.463229   
3995                Chrást nad Sázavou 239 - parkoviště  49.847983  14.551752   
4009                          Za Humny - naproti Horská  50.642080  14.001179   
4050  Pařížská - vedle novostavby řadových garáží př...  50.135930  14.135830   
4173                                  Aventin - Okružní  49.044367  15.803005   
4262                      parkoviště Slánská - Žufanova  50.069942  14.309688   
4264                   zastávka Bílá Hora - Karlovarská  50.075387  14.321271   

     house_no  
1909      NaN  
2864      NaN  
2870      511  
2960      NaN  
3096      NaN  
3106      NaN  
3133      NaN  
3207      NaN  
3466      NaN  
3532      NaN  
3775  1484/2a  
3995      NaN  
4009    211/2  
4050       17  
4173     1779  
4262   1097/6  
4264       25

pro zjednodušení odstranit určité fráze

In [28]:
specificke_adresy = df[df['address'].str.contains(' - parkoviště| - u myček| - Parkoviště| - za automyčkou| - vedle katastrálního úřadu| - vjezd z ulice Jizbická', na=False)]

df.loc[:,'address'] = df['address'].str.replace(' - parkoviště| - u myček| - Parkoviště| - za automyčkou| - vedle katastrálního úřadu| - vjezd z ulice Jizbická', '', regex = True)
df.columns
specificke_adresy

name                      city zipcode  \
2864  GLS Parcel Box        Veselí nad Lužnicí   39181   
2960  GLS Parcel Box                  Litomyšl   57001   
3106  GLS Parcel Box                    Náchod   54701   
3133  GLS Parcel Box       Rychnov nad Kněžnou   51601   
3466  GLS Parcel Box                 Nová Paka   50901   
3532  GLS Parcel Box  Praha 20 Horní Počernice   19300   
3995  GLS Parcel Box        Chrást nad Sázavou   25742   

                                           address     geolat     geolng  \
2864       Tř. Československé Armády 609 - u myček  49.197283  14.716687   
2960                Sokolovská 113 - za automyčkou  49.880838  16.292915   
3106                          Mlýnská - parkoviště  50.414193  16.162277   
3133      Zborovská 17 - vedle katastrálního úřadu  50.163417  16.278833   
3466                    Na Tržišti 23 - Parkoviště  50.494030  15.515213   
3532  Pavlišovská 2292/14 - vjezd z ulice Jizbická  50.110422  14.621840   
3995           Chrást nad Sázavou 239 - parkoviště  49.847983  14.551752   

     house_no  
2864      NaN  
2960      NaN  
3106      NaN  
3133      NaN  
3466      NaN  
3532      NaN  
3995      NaN

opravit cca 10 řádků, kde je cela adresa v řádku neoddělená
(Lhotská 12, 756 25 - Ostrava)
pro jednoduchost odstranim vse za carkou, jedná se o 10 řádků

In [29]:
# adresy_s_carkou = df[df['address'].str.contains(',', na=False)]
# address_text_rozdelene = df['address'].str.rsplit(',', n=1, expand=True)

# df.loc[:,'zahodit'] = address_text_rozdelene[1].str.strip()
# df.loc[:,'address'] = address_text_rozdelene[0].str.strip()

# df.head(2)

Dočistím řádky, kde ještě zůstává č.p. 

In [30]:
df.to_csv('gls_pomocny_dataset.csv', index=False, encoding='utf-8')